### PR TITLE
Specify types for all variables and pull in upstream updates

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request adds types for all variables.  I also pull in upstream updates while I'm at it.

## 💭 Motivation and Context

I am going through all the Terraform module repositories to remove non-`alias` variables from proxy providers.  This repository did not require that change, but I noticed that the variable types were not always specified.

## 🧪 Testing

All pre-commit hooks pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
LGTM! 🐐

File: README.md [hunk 1]
```diff
@@ -71,6 +84,12 @@ module "ipa_master" {
 | master | The IPA master EC2 instance. |
 | server_security_group | The IPA server security group. |
 
+## Notes ##
+
+Running `pre-commit` requires running `terraform init` in every directory that
+contains Terraform code. In this repository, these are the main directory and
+every directory under `examples/`.
+
```
Comment (mcdonnnj) [inline README.md]: I believe you would need to run `terraform init` in the `dns` directory as well.

**Review by mcdonnnj — APPROVED** [review]
Lookin' good 👍, but please see my one comment about the README.

File: README.md (hunk 1 above)
Comment (jsf9k) [reply README.md]: See commit 4a71226.

Comment (mcdonnnj) [reply README.md]: Thanks as always, but a typo emerged in that commit: "a well as" -> "as well as" (sorry).

Comment (jsf9k) [reply README.md]: See commit 5c8350f.

Comment (mcdonnnj) [reply README.md]: Appreciate you 👍

**Pull request merged**